src/vs/platform/quickinput/common/quickInput.ts

```diff
@@ -94,4 +95,10 @@ export interface IQuickInputService {
 	 * Cancels quick input and closes it.
 	 */
 	cancel(): Promise<void>;
+
+	/**
+	 * Formats quick pick items to be pinnable
+	 * with a sticky pinned category at the top
```

Comment:
We should explain the params here and the quirks of the function:

- What happens when multiple labels exist; only the first gets pinned
- The original item isn't moved but a duplicate pinned item is added
- etc.